coordinator/src/main/scala/com/socrata/datacoordinator/resources/NotFoundDatasetResource.scala

```diff
@@ -0,0 +1,112 @@
+import com.rojoma.json.v3.util.JsonArrayIterator
+import com.rojoma.simplearm.util._
+import com.socrata.datacoordinator.external.BodyRequestError
+import com.socrata.datacoordinator.id.DatasetId
+import com.socrata.datacoordinator.service.{Service, ProcessCreationReturns}
+import com.socrata.datacoordinator.service.ServiceUtil._
+import com.socrata.datacoordinator.util.IndexedTempFile
+import com.socrata.http.server._
+import com.socrata.http.server.responses._
+import org.joda.time.format.ISODateTimeFormat
+import com.socrata.http.server.implicits._
+
+
+case class NotFoundDatasetResource(datasetIdRaw: Option[String],
+                                   formatDatasetId: DatasetId => String,
+                                   tempFileProvider: () => IndexedTempFile,
+                                   processCreation: (Iterator[JValue], IndexedTempFile) => ProcessCreationReturns,
+                                   datasets: () => Seq[DatasetId],
+                                   withMutationScriptResults: (HttpResponse) => HttpResponse,
+                                   commandReadLimit: Long)
+  extends ErrorHandlingSodaResource(formatDatasetId) {
+
+
+  override val log = org.slf4j.LoggerFactory.getLogger(classOf[NotFoundDatasetResource])
+  val dateTimeFormat = ISODateTimeFormat.dateTime
+
+  override def post = {
+    datasetIdRaw match {
+      case None => _:HttpRequest => notFoundError("")
+      case Some(datasetId) => doCreateDataset
+    }
+  }
+  override def delete = {
+    datasetIdRaw match {
+      case None => _:HttpRequest => notFoundError("")
+      case Some(datasetId) => _:HttpRequest =>notFoundError(datasetId)
+    }
+  }
+
+  override def get = {
+    datasetIdRaw match {
+      case None => _:HttpRequest => notFoundError("")
+      case Some(datasetId) => doListDatasets
+    }
+  }
+
+
+  def doCreateDataset(req: HttpRequest)(resp: HttpServletResponse) {
+    using(tempFileProvider()) { tmp =>
+      val responseBuilder = withMutationScriptResults {
+        jsonStream(req.servletRequest, commandReadLimit) match {
+          case Right((events, boundResetter)) =>
+            val iteratorOrError = try {
+              Right(JsonArrayIterator[JValue](events))
+            } catch { case _: JsonBadParse =>
+              Left(datasetErrorResponse(BadRequest, BodyRequestError.NOT_JSON_ARRAY))
+            }
+            iteratorOrError match {
+              case Right(iterator) =>
+                val ProcessCreationReturns(dataset, dataVersion, lastModified, result) = processCreation(iterator.map { ev => boundResetter(); ev }, tmp)
+                OK ~>
+                  Header("X-SODA2-Truth-Last-Modified", dateTimeFormat.print(lastModified)) ~>
+                  Header("X-SODA2-Truth-Version", dataVersion.toString) ~>
+                  ContentType(JsonContentType) ~>
+                  Stream { w =>
+                    val bw = new BufferedOutputStream(w)
```

Comment:
Of course it can, but we're not building an AST here; we're streaming a result.  We absolutely don't want to ever put `result` in a `JArray` as that would hold on to the whole thing in memory.

In modern rojoma-json, it might be better to use an `ArrayIteratorEncoder` (from `c.r.j.v3.util`) here, but that didn't exist when this code (which is being moved from elsewhere) was written.